packages/meditrak-server/src/routes/exportSurveyResponses/exportSurveyResponses.js

```diff
@@ -143,45 +149,54 @@ export async function exportSurveyResponses(req, res) {
         const entity = await models.entity.findById(surveyResponse.entity_id);
         await processSurveyResponse(surveyResponse, entity);
       } else {
-        for (let entityIndex = 0; entityIndex < entities.length; entityIndex++) {
-          const entity = entities[entityIndex];
-          const surveyResponseFindConditions = {
-            survey_id: currentSurvey.id,
-            entity_id: entity.id,
-          };
-          if (startDate && endDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparisonType: 'whereBetween',
-              args: [
-                [
-                  new Date(moment(startDate).subtract(1, 'day')),
-                  new Date(moment(endDate).add(1, 'day')),
-                ],
+        const surveyResponseFindConditions = {
+          survey_id: currentSurvey.id,
+        };
+        if (startDate && endDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparisonType: 'whereBetween',
+            args: [
+              [
+                new Date(moment(startDate).subtract(1, 'day')),
+                new Date(moment(endDate).add(1, 'day')),
               ],
-            };
-          } else if (startDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparator: '>=',
-              comparisonValue: new Date(startDate),
-            };
-          } else if (endDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparator: '<=',
-              comparisonValue: new Date(endDate),
-            };
-          }
-          const surveyResponses = await models.surveyResponse.find(
-            surveyResponseFindConditions,
-            sortAndLimitSurveyResponses,
-          );
+            ],
+          };
+        } else if (startDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparator: '>=',
+            comparisonValue: new Date(startDate),
+          };
+        } else if (endDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparator: '<=',
+            comparisonValue: new Date(endDate),
+          };
+        }
+        const entityIdsGroup = entities.map(entity => entity.id);
+        surveyResponseFindConditions.entity_id = entityIdsGroup;
 
-          for (
-            let surveyResponseIndex = 0;
-            surveyResponseIndex < surveyResponses.length;
-            surveyResponseIndex++
-          ) {
-            await processSurveyResponse(surveyResponses[surveyResponseIndex], entity);
-          }
+        const surveyResponses = await models.surveyResponse.find(
+          surveyResponseFindConditions,
+          sortAndLimitSurveyResponses,
+        );
+        const nestedEntitiesById = keyBy(entities, 'id');
```

Comment:
`keyBy` maps an item to another item, so I wouldn't think that entities are nested here

```suggestion
        const entitiesById = keyBy(entities, 'id');
```